### PR TITLE
feat: add support for key vault KVContribtors

### PR DESCRIPTION
**Pull request by ddlees**

### CODE REVIEW

File: cmd/list-key-vault-contributors.go [hunk 1]
```diff
@@ -58,75 +57,52 @@ func listKeyVaultContributorsCmdImpl(cmd *cobra.Command, args []string) {
...
 }
 
-func listKeyVaultContributors(ctx context.Context, client client.AzureClient, KeyVaults <-chan interface{}) <-chan interface{} {
-	var (
-		out     = make(chan interface{})
-		ids     = make(chan string)
-		streams = pipeline.Demux(ctx.Done(), ids, 25)
-		wg      sync.WaitGroup
-	)
+func listKeyVaultContributors(ctx context.Context, client client.AzureClient, vmRoleAssignments <-chan interface{}) <-chan interface{} {
```
Comment (irshadaj) [inline cmd/list-key-vault-contributors.go]: We aren't using the `client` param any more, I think we should remove it.

File: cmd/list-key-vault-kvcontributors.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		keyVaults := listKeyVaults(ctx, azClient, subscriptions)
+		kvRoleAssignments := listKeyVaultRoleAssignments(ctx, azClient, keyVaults)
+		stream := listKeyVaultKVContributors(ctx, azClient, kvRoleAssignments)
+		outputStream(ctx, stream)
+		duration := time.Since(start)
+		log.Info("collection completed", "duration", duration.String())
+	}
+}
+
+func listKeyVaultKVContributors(ctx context.Context, client client.AzureClient, vmRoleAssignments <-chan interface{}) <-chan interface{} {
```
Comment (irshadaj) [inline cmd/list-key-vault-kvcontributors.go]: I'd suggest combining `listKeyVaultContributors` and `listKeyVaultKVContributors` into a single function, with an additional `contributorRoleID` parameter to differentiate

File: cmd/list-key-vault-kvcontributors.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+var listKeyVaultKVContributorsCmd = &cobra.Command{
+	Use:          "key-vault-kvcontributors",
+	Long:         "Lists Azure Key Vault KVContributors",
+	Run:          listKeyVaultKVContributorsCmdImpl,
+	SilenceUsage: true,
+}
+
+func listKeyVaultKVContributorsCmdImpl(cmd *cobra.Command, args []string) {
```
Comment (irshadaj) [inline cmd/list-key-vault-kvcontributors.go:45]: `args` is unused here
Also the same comment as below- this and `listKeyVaultContributorsCmdImpl` should be dried out into a single function

File: cmd/list-key-vault-owners.go [hunk 4]
```diff
@@ -58,75 +57,52 @@ func listKeyVaultOwnersCmdImpl(cmd *cobra.Command, args []string) {
...
 }
 
-func listKeyVaultOwners(ctx context.Context, client client.AzureClient, keyVaults <-chan interface{}) <-chan interface{} {
-	var (
-		out     = make(chan interface{})
-		ids     = make(chan string)
-		streams = pipeline.Demux(ctx.Done(), ids, 25)
-		wg      sync.WaitGroup
-	)
+func listKeyVaultOwners(ctx context.Context, client client.AzureClient, vmRoleAssignments <-chan interface{}) <-chan interface{} {
```
Comment (irshadaj) [inline cmd/list-key-vault-owners.go]: `client` is unused

File: cmd/list-key-vault-kvcontributors.go (hunk 2 above)
Comment (ddlees) [reply cmd/list-key-vault-kvcontributors.go]: Those two functions emit different types in the output channel so I can't combine them cleanly but I can probably introduce a generic `Filter` and `Map` function for channels to keep things DRY.

**Pull request merged**